twill-api/src/main/java/org/apache/twill/api/EventHandler.java

```diff
@@ -124,6 +124,75 @@ public void initialize(EventHandlerContext context) {
   }
 
   /**
+   * Invoked by the application when it starts.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   */
+  public void started(String twillAppName, RunId runId) {
```

Comment:
Why all methods has the app name and runid? It should be provided via the `EventHandlerContext` since it will never change for an application run.